test/gtest/uct/uct_test.cc

```diff
@@ -961,8 +961,14 @@ void uct_test::entity::mem_alloc(size_t length, unsigned mem_flags,
         status = uct_mem_alloc(length, alloc_methods,
                                ucs_static_array_size(alloc_methods), &params,
                                mem);
+    }
+
+    if (may_fail && (status != UCS_OK)) {
```

Comment:
Do we really need to pass `may_fail` param, can we just catch exception from ASSERT_UCS_OK?

Reply:
Yes, we need to pass `may_fail` because of the way `ASSERT_UCS_OK` handles failures, it does more than just throwing exception, it collects the failures to mark the test as successful or not at the end of the test.